lib/controllers/rooms_controller.dart

```
@@ -142,4 +145,32 @@ class RoomsController extends GetxController {
       Get.back();
     }
   }
+
+  Future<void> searchLiveRooms(String query) async {
+    if (query.isEmpty) {
+      filteredRooms.value = rooms;
+      searchBarIsEmpty.value = true;
+      return;
+    }
+    isSearching.value = true;
+    searchBarIsEmpty.value = false;
+    try {
+      final lowerQuery = query.toLowerCase();
+      filteredRooms.value = rooms.where((room) {
+        return room.name.toLowerCase().contains(lowerQuery) ||
+            room.description.toLowerCase().contains(lowerQuery);
+      }).toList();
+    } catch (e) {
+      log('Error searching rooms: $e');
```

Instead of a silent failure, reset the searched rooms to all rooms and display an error message